### PR TITLE
Asher meta merge

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

I tried to sort out the meta merge issue in pull #302 via the directions. I made my own branch to make sure I didn't mess things up.

Addresses # (issue)

Pull #302 

## Type of change

# How Has This Been Tested?

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by rstoneback** [comment]
unsure if this helps or hurts.

@asher-pembroke and @jklenzing

**Comment by asher-pembroke** [comment]
Thanks for doing this. I'm still getting used to the merging procedure and keep getting confused on which branch to merge to/from.

**Comment by rstoneback** [comment]
No worries.

FYI we are getting a small test error.

======================================================================
FAIL: pysat.tests.test_meta.TestBasics.test_transfer_attributes_to_instrument_no_meta_attr
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/nose/tools/nontrivial.py", line 67, in newfunc
    raise AssertionError(message)
AssertionError: test_transfer_attributes_to_instrument_no_meta_attr() did not raise AttributeError

**Comment by rstoneback** [comment]
Something has gone wrong with python 3

**Comment by jklenzing** [comment]
😱

**Comment by jklenzing** [comment]
I've not checked every error, but most of them seem to have:
```
  File "/home/travis/build/pysat/pysat/pysat/_instrument.py", line 391, in __getattr__
    return getattr(self.meta, name)
  File "/home/travis/build/pysat/pysat/pysat/_instrument.py", line 391, in __getattr__
    return getattr(self.meta, name)
  File "/home/travis/build/pysat/pysat/pysat/_instrument.py", line 391, in __getattr__
    return getattr(self.meta, name)
  [Previous line repeated 468 more times]
RecursionError: maximum recursion depth exceeded in comparison
```

**Comment by jklenzing** [comment]
I see two different versions of `getattr` in the two metadata pulls.  Going to try replacing the recursive version with the one from #302.

**Comment by jklenzing** [comment]
That fixed 3.7, but 2.7 is crashing now.  Not sure what is going on there, it starts the tests but then pauses almost immediately and times out.  Travis is undergoing some issues where it's not letting me restart builds for orgs.

**Comment by jklenzing** [comment]
It's not just me having this issue.  is anyone else running into it?

https://travis-ci.community/t/suddently-travis-ci-says-that-im-not-a-member-of-any-organization/5593

**Comment by rstoneback** [comment]
Looks like my merge went astray somehow. 
That sucks. Sorry about that.

**Comment by jklenzing** [comment]
Travis issues seem to be related to the push to get users to migrate to travis-ci.com.  Migrated sami2py over there, and the org access works as expected.

**Comment by jklenzing** [comment]
2.7 crashes in both the push and pr on travis.  My guess is that something in the `getattr` is not 2.7 compliant, so it hangs the first time an instrument is run.

**Comment by asher-pembroke** [comment]
I'll try to make getattr work with 2.7

**Comment by asher-pembroke** [comment]
pysat is hanging on 2.7 when I try to make a test instrument

**Comment by asher-pembroke** [comment]
The tests seem to be working on my machine with 2.7 now. Going to see what travis thinks

File: docs/new_instrument.rst [hunk 1]
```diff
@@ -261,7 +261,7 @@ as a tag to delineate that the data contains the UTD developed quality flags.
    sat_ids = {'f11': ['utd', ''], 'f12': ['utd', ''], 'f13': ['utd', ''],
               'f14': ['utd', ''], 'f15': ['utd', ''], 'f16': [''], 'f17': [''],
               'f18': ['']}
-   test_dates = {'f11': {'utd': pysat.datetime(1998, 1, 2)},
+   _test_dates = {'f11': {'utd': pysat.datetime(1998, 1, 2)},
                  'f12': {'utd': pysat.datetime(1998, 1, 2)},
                  'f13': {'utd': pysat.datetime(1998, 1, 2)},
                  'f14': {'utd': pysat.datetime(1998, 1, 2)},
```
Comment (jklenzing) [inline docs/new_instrument.rst]: Continued lines should be indented one more space

**Comment by jklenzing** [comment]
Also, should we close #302 in favor of this one?

**Comment by rstoneback** [comment]
whew. There sure is a bunch of code changes here. 

@jklenzing the cnofs code is flagged as new as well. Is the code listed here the correct code?

**Comment by jklenzing** [comment]
> whew. There sure are a bunch of code changes here.
> 
> @jklenzing the cnofs code is flagged as new as well. Is the code listed here the correct code?

Yes, those are the latest changes.

**Pull request merged**